recidiviz/ingest/models/ingest_info.py

```
@@ -17,34 +17,44 @@
 
 """Represents data scraped for a single individual."""
 
-from recidiviz.common.constants.bond import BondStatus
-from recidiviz.common.constants.bond import BondType
-from recidiviz.common.constants.booking import Classification
-from recidiviz.common.constants.booking import CustodyStatus
-from recidiviz.common.constants.booking import JurisdictionStatus
-from recidiviz.common.constants.booking import ReleaseReason
-from recidiviz.common.constants.charge import ChargeClass
-from recidiviz.common.constants.charge import ChargeStatus
-from recidiviz.common.constants.charge import CourtType
-from recidiviz.common.constants.charge import Degree
-from recidiviz.common.constants.person import Ethnicity
-from recidiviz.common.constants.person import Race
+# from recidiviz.common.constants.bond import BondStatus
+# from recidiviz.common.constants.bond import BondType
+# from recidiviz.common.constants.booking import Classification
+# from recidiviz.common.constants.booking import CustodyStatus
+# from recidiviz.common.constants.booking import JurisdictionStatus
+# from recidiviz.common.constants.booking import ReleaseReason
+# from recidiviz.common.constants.charge import ChargeClass
+# from recidiviz.common.constants.charge import ChargeStatus
+# from recidiviz.common.constants.charge import CourtType
+# from recidiviz.common.constants.charge import Degree
+# from recidiviz.common.constants.person import Ethnicity
+# from recidiviz.common.constants.person import Race
 
 
 class IngestInfo(object):
     """Class for information about multiple people."""
 
     def __init__(self):
-        self.people = []  # type: List[Person]
+        self.person = []  # type: List[Person]
```

Why the singular name for the array type?

---

If you look at the extractor code, this lets us avoid any annoying mappings required in the data extractor, and lets us dynamically find objects by their parents using the names.  The code would otherwise be a lot more annoying to write as we'd have to keep some type of mapping between whats in the yaml, and what exists internally.

---

Ah, I understand. It's working because it's coupling the `get_recent_*` method names to the names of the entity classes themselves. That's clever but let's make sure we've got good test coverage/documentation such that if any names change on either end, we don't silently fail.

---

Agreed.